tools/nbs_tools/calibrate_cameras.py

```diff
@@ -0,0 +1,507 @@
+#!/usr/bin/env python3
+
+import bisect
+import copyreg
+import json
+import math
+import multiprocessing
+import os
+import pickle
+
+import cv2
+import numpy as np
+import tensorflow as tf
+from ruamel.yaml import YAML
+from tqdm import tqdm
+
+import b
+
+from .camera_calibration.callback import EpochProgress
+from .camera_calibration.loss import ExtrinsicLoss, euclidean_error, euclidean_loss
+from .camera_calibration.metric import GridError, alignment, collinearity, orthogonality, parallelity
+from .camera_calibration.model import Equidistant, Equisolid, ExtrinsicCluster, Rectilinear
+from .decoder import Decoder
+from .images import decode_image, fourcc
+
+ROWS = 4
+COLS = 11
+
+
+def _pickle_keypoints(point):
+    return cv2.KeyPoint, (*point.pt, point.size, point.angle, point.response, point.octave, point.class_id)
+
+
+copyreg.pickle(cv2.KeyPoint().__class__, _pickle_keypoints)
```

Comment:
Good idea, this one is because OpenCV types are weird and not pickleable which I need to do to use multiprocessing. This overwrites something in OpenCV so the types can be pickled properly.